src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpSystemProxy.cs

```diff
@@ -131,8 +156,59 @@ public Uri GetProxy(Uri uri)
         {
             if (_proxyHelper.ManualSettingsOnly)
             {
+                if ( _bypassLocal)
+                {
+                    IPAddress address = null;
+
+                    if (uri.IsLoopback)
+                    {
+                        // This is optimization for loopback addresses.
+                        // Unfortunately this does not work for all local addresses.
+                        return null;
+                    }
+                    if (uri.Host[0] == '[' || Char.IsNumber(uri.Host[0]))
+                    {
+                        // RFC1123 allows labels to start with number.
+                        // Leading number may or may not be IP address.
+                        // IPv6 [::1] notation. '[' is not valid character in names.
+                        IPAddress.TryParse(uri.Host, out address);
+                    }
+                    if (address != null)
+                    {
+                        // Host is valid IP address.
+                        // Check if it belongs to local system.
+                        foreach (var a in _localIp)
+                        {
+                            if (a.Equals(address))
+                            {
+                                return null;
+                            }
+                        }
+                    }
+                    else
+                    {
+                        // Hosts without FQDN are considered local.
+                        if (uri.Host.IndexOf('.') == -1) return null;
+                    }
+                }
+
+                // Check if we have other rules for bypass.
+                if (_bypass != null)
+                {
+                    foreach (var entry in _bypass)
+                    {
+                        if (entry.IsMatch(uri.Host))
+                        {
+                            return null;
+                        }
+                    }
+                }
+
+                // We did not find match on bypass list.
```

Comment:
Am I correct in understanding that this code is going to be run on every request made to a default-configured SocketsHttpHandler?  Have you looked to see what if any perf impact it has?

Reply:
the match should only run when  there is something on the bypass list.
As we discussed today @stephentoub , it seems like "auto" is default so we will not get to the manual processing unless somebody asks for it.